tests/unit/s2n_self_talk_ktls_test.c

```diff
@@ -87,11 +90,15 @@ static S2N_RESULT s2n_new_inet_socket_pair(struct s2n_test_io_pair *io_pair)
     RESULT_ENSURE_GTE(pid, 0);
     if (pid == 0) {
         RESULT_ENSURE_EQ(connect(io_pair->client, (struct sockaddr *) &saddr, addrlen), 0);
+        EXPECT_SUCCESS(s2n_io_pair_close(io_pair));
         ZERO_TO_DISABLE_DEFER_CLEANUP(io_pair);
+        close(listener);
```

Comment:
> Could the listener socket be created after the fork to avoid the additional close?

I don't think so. The child process needs `saddr` to be fully set up, so we need `listener` to be there before the fork for set up purpose. We use [`bind`](https://man7.org/linux/man-pages/man2/bind.2.html),[`getsocketname`](https://man7.org/linux/man-pages/man2/getsockname.2.html), and [`listen`](https://man7.org/linux/man-pages/man2/listen.2.html) to set up `saddr` which all depends on `listener`.